### ISSUE TITLE
Stepper: changing the "numbering-systems" attribute via dev tools does not cause page to re-render with updated numbering system

### ISSUE THREAD

**Issue by annierm18** [Esri/calcite-components#5979]
### Actual Behavior

In Stepper, changing the "numbering-systems" attribute via dev tools (not via the code in codepen), does not cause the page to re-render with the updated numbering systems. 

![Screenshot (108)](https://user-images.githubusercontent.com/25759835/206820929-a731eb6e-f217-484e-86d9-6d1bf6011a20.png)

### Expected Behavior

When a user changes the numbering system, the component should update and show the new numbering system.

### Reproduction Sample

https://codepen.io/ANVESHMEKALA/pen/NWzmpdd?editors=1010

### Reproduction Steps

1. Open either Stepper in codepen 
2. Open the dev tools. 
3. Change the numbering-system to a non-latn numbering system

### Reproduction Version

http://calcite-dev.esri.com:3008/demos/input-time-picker.html

### Esri team

N/A

**Comment by benelan** [comment]
FWIW, changing `numberingSystem` in the code doesn't update it either; it's not specific to dev tools:
https://codepen.io/benesri/pen/OJEGgzY?editors=1010

However, it will rerender if you also change `lang` afterwards (uncomment line 3 in the JS section of the codepen above). 

What is the use case for dynamically changing the numbering system and not `lang`? When I spoke to Franco about this we couldn't think of one other than testing. See: https://github.com/Esri/calcite-components/pull/5467#discussion_r995020457

At minimum our doc needs to be updated with this information though.

**Comment by annierm18** [comment]
@benelan Good point, this issue seems mostly applicable to testing. cc: @anveshmekala

**Comment by anveshmekala** [comment]
I think what @annierm18  meant is when she changes the prop in codepen HTML page it re-renders the component and hence updated numbering system is displayed.

The re-render happens only when the `lang` prop is changed not the numbering system.

Regarding use case , if the app has to show an option of switching between two different numbering systems for a given lang (ex: Arabic locale). I agree we have to decide if it's the most common use-case with Esri applications at least to make the change.

**Comment by benelan** [comment]
Ah gotcha. Yeah that's Codepen's auto-reload doing that but if you turn it off it won't re-render.

> Regarding use case , if the app has to show an option of switching between two different numbering systems for a given lang (ex: Arabic locale). I agree we have to decide if it's the most common use-case with Esri applications at least to make the change.

Yeah, I don't think that uncommon use case is worth the need to add another MutationObserver or internal event though. In my opinion we should wait to see if any users log an issue or provide feedback about it.I'll defer to Franco on this one when he gets back. But until then we do need a doc update, cc @geospatialem

**Comment by geospatialem** [comment]
This has been updated in the docs site on the Localization page for the January 1.0 release. 

Will leave this issue open for the February milestone to get Franco's feedback if next steps are needed.

**Comment by jcfranco** [comment]
@benelan For behavior consistency, we could force items to render when the stepper's `numberingSystem` changes (without using a mutation observer). Something [like this](https://github.com/Esri/calcite-components/pull/6563). This is the only parent-child component where only the child uses `numberingSystem` for rendering, right? If so, maybe we need to make a note of it since it's a special case.

**Comment by geospatialem** [comment]
Re-allocating to the upcoming March milestone.

**Comment by geospatialem** [comment]
Remaining item for implementation is a test to support the `stepper` component.
